error.go

```diff
@@ -7,12 +7,20 @@ import (
 )
 
 // APIError provides error information returned by the OpenAI API.
+// InnerError struct is only valid for Azure OpenAI Service.
 type APIError struct {
-	Code           any     `json:"code,omitempty"`
-	Message        string  `json:"message"`
-	Param          *string `json:"param,omitempty"`
-	Type           string  `json:"type"`
-	HTTPStatusCode int     `json:"-"`
+	Code           any        `json:"code,omitempty"`
+	Message        string     `json:"message"`
+	Param          *string    `json:"param,omitempty"`
+	Type           string     `json:"type"`
+	HTTPStatusCode int        `json:"-"`
+	InnerError     InnerError `json:"innererror,omitempty"`
```

Comment:
@ZeroDeng01  Since InnerError is optional, please make it a pointer.

Reply:
@vvatanabe Hi, this is a great idea, I didn't think of it at first, it has been revised, THANSK! Please help with the review!